### ISSUE TITLE
Server dies with Prisma error when there are no entities defined

### ISSUE THREAD

**Issue by shayneczyzewski** [wasp-lang/wasp#398]
**Describe the bug**
Server dies with Prisma error when the `main.wasp` file doesn't specify any entities.

**To Reproduce**
1. `wasp new prismaTest`
2. Add ext/static.js file:
```
export const getStaticData = async (args, context) => {
  return { foo: 123 }
}
```
3. Update main.wasp to use it:
```
query getStaticData {
  fn: import { getStaticData } from "@ext/static.js"
}
```
4. `wasp start` produces the following error:
```
Server: /private/tmp/prismatest/.wasp/out/server/node_modules/.prisma/client/index.js:3
Server (stderr):     throw new Error(
Server (stderr):           ^
Server (stderr):
Server (stderr): Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
Server (stderr): In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
Server (stderr):     at new PrismaClient (/private/tmp/prismatest/.wasp/out/server/node_modules/.prisma/client/index.js:3:11)
Server (stderr):     at createDbClient (file:///private/tmp/prismatest/.wasp/out/server/src/dbClient.js:4:24)
Server (stderr):     at file:///private/tmp/prismatest/.wasp/out/server/src/dbClient.js:9:18
Server (stderr):     at ModuleJob.run (internal/modules/esm/module_job.js:152:23)
Server (stderr):     at async Loader.import (internal/modules/esm/loader.js:166:24)
Server (stderr):     at async Object.loadESM (internal/process/esm_loader.js:68:5)
Server (stderr): [nodemon] app crashed - waiting for file changes before starting...
Server:
```

<img width="1002" alt="Screen Shot 2021-12-13 at 10 54 44 AM" src="https://user-images.githubusercontent.com/523636/145844724-623f2e0c-252a-4dd9-bc6d-8ef51aa5f2e4.png">

**Expected behavior**
Allow users to use create apps without any Entities defined (e.g., in-memory/static server apps).

**Desktop (please complete the following information):**
 - OS: macOS 12.0.1
 - Wasp: 0.2.2.2

**Additional context**
The fix for now is to define a dummy Entity in main.wasp and run `wasp db migrate-dev` on it:
```
entity Dummy {=psl
    id          Int     @id @default(autoincrement())
psl=}
```

**Comment by sodic** [comment]
Make sure the following features work with this as well:
- #902

**Comment by shayneczyzewski** [comment]
> 

I agree, it would be good to fix this. Although I think I took a quick look a long time ago, and it was a bit more complicated than it first seemed, and @sodic did recently too and also bailed. It is not the import I believe, but rather when we call `prisma generate` with no entities, amongst maybe other assumptions (possibly imports after that initial issue).

**Comment by sodic** [comment]
Yeah, I can't remember what it was (should have written it down), but Shayne's right. I thought it was gonna be a 1-day task and then concluded that it's going to take longer (maybe not much longer, but I couldn't spare more than a day).

That said, I bailed pretty early so you can definitely try to take a stab at it (the obstacles Shayne and I encountered should become apparent right away).

**Comment by shayneczyzewski** [comment]
So @infomiho and I spent our pairing today looking into this, and the good news is this is no longer a problem! 🥳 

Thanks to the shift to TS (@sodic 👋🏻), any place where we imported `dbClient` but didn't use, it now gets removed. So in our current implementation, we only will actually have those imports in the dist output when we have entities (and thus, a Prisma client can/will be generated).

We checked all the files importing `dbClient` to make sure this is true, and the only place this isn't true is for DB seeding @Martinsos. This isn't a huge problem, since most people will only run that when they have entities, but if we wanted to we could add some checks before invoking that command to warn if they don't have entities.

We could also more generally improve the checks/errors resulting from trying to create the Prisma client in `dbClient` in the future as well. But the long story short is, we should not hit this error anymore due to unused import removal. 👍🏻
